stdlib/random.ml

```diff
@@ -40,6 +40,37 @@ module State = struct
     let s = create () in
     set s i1 i2 i3 i4; s
 
+  let serialization_prefix =
+    (* '1' is a version number; more than 200 versions may fit a char. *)
+    "lxm1:"
+  let serialization_prefix_len =
+    String.length serialization_prefix
+
+  let to_binary_string s =
+    let prefix = serialization_prefix in
+    let preflen = serialization_prefix_len in
+    let buf = Bytes.create (preflen + 4 * 8) in
+    Bytes.blit_string prefix 0 buf 0 preflen;
+    for i = 0 to 3 do
+      Bytes.set_int64_le buf (preflen + i * 8) (Array1.get s i)
+    done;
+    Bytes.unsafe_to_string buf
+
+  let of_binary_string buf =
+    let prefix = serialization_prefix in
+    let preflen = serialization_prefix_len in
+    if String.length buf <> preflen + (4 * 8)
+       || not (String.starts_with ~prefix buf)
+    then
+      Printf.ksprintf failwith
+        "Random.State.of_binary_string: invalid format (expected version: %s)"
+        Sys.ocaml_version;
```

Comment:
However since multiple ocaml versions are going to share the same formt this is slightly misleading. Why not use:
```
  "Random.State.of_binary_string: invalid format (expected %s)" serialization_prefix;
```

Reply:
Thanks for the feedback.

My intuition is that users have no clue about the serialization format they are producing and/or should be expecting, but that they may have a sense of which OCaml version they expect to be running, and understand what is going on if they see a different version from that.

Maybe we could say "expected a format compatible with {version}"?

This being said:
- I'm making the assumption that the issue is related to the version of the running program. (The code could be incorrect in other ways, or consume serialized data from another version, etc.)
- The hint also impresses this assumption into the users and may lead them in false directions
- I don't have the energy to find the best approach right now, so I think that I will just fall back to the previous implementation. (Also: maybe I will be able to build the branch locally again.)

Reply:
> Maybe we could say "expected a format compatible with {version}"?

Maybe that's good enough. That way you can also simply:
```
failwith ("Random.State.of_binary_string: expected a format compatible with OCaml " ^ Sys.ocaml_version)
```
and get rid of our `.depend` woes.

Reply:
Barely, thanks to the fact that Int32 and Nativeint already depend on Sys. Phew! Thanks, I will take your suggestion.